From 2bff4b6fbecf87f91bce64ad99e94540ed9f9f51 Mon Sep 17 00:00:00 2001
From: Blue Swirl <blauwirbel@gmail.com>
Date: Wed, 23 Dec 2009 15:34:04 +0000
Subject: [PATCH] Compile qemu-nbd also on OpenBSD and Solaris

basename() needs #include <libgen.h>.

No prototype for daemon() is available on Solaris, but link
succeeds.

Signed-off-by: Blue Swirl <blauwirbel@gmail.com>
---
 configure  | 2 +-
 qemu-nbd.c | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/qemu-nbd.c b/qemu-nbd.c
index 6cdb83473f9..91ea01414d1 100644
--- a/tools/qemu-xen-traditional/qemu-nbd.c
+++ b/tools/qemu-xen-traditional/qemu-nbd.c
@@ -30,6 +30,7 @@
 #include <netinet/tcp.h>
 #include <arpa/inet.h>
 #include <signal.h>
+#include <libgen.h>
 
 #define SOCKET_PATH    "/var/lock/qemu-nbd-%s"
 
diff -rup pci.c.orig pci.c
--- a/tools/qemu-xen/hw/vfio/pci.c
+++ b/tools/qemu-xen/hw/vfio/pci.c
@@ -21,6 +21,7 @@
 #include "qemu/osdep.h"
 #include <linux/vfio.h>
 #include <sys/ioctl.h>
+#include <libgen.h>
 
 #include "hw/hw.h"
 #include "hw/pci/msi.h"

